pandas/compat/__init__.py

```diff
@@ -74,11 +74,9 @@ def raise_with_traceback(exc, traceback=Ellipsis):
     raise exc.with_traceback(traceback)
 
 
-# In Python 3.7, the private re._pattern_type is removed.
-# Python 3.5+ have typing.re.Pattern
```

Comment:
I removed these comments because they're misleading. `typing.re.Pattern` is available in 3.5-3.7. It's been removed from the documentation as of 3.6, but it continues to be available. As of 3.6, the documented method is to use `typing.Pattern`, which also works in 3.5. I've adopted that pattern in this PR, which corrected the mypy error